### PR TITLE
Per Monitor V2 DPI aware support

### LINKED ISSUE #1239: Per Monitor V2 DPI awareness Support?
### Description of the feature, modification, idea or suggestion

Not sure an introduction is necessary, but: Per-monitor V2 DPI awareness would allow process hacker to seamlessly be dragged across monitors with different DPI settings without getting stuck blurry compositor upscaled like currently. This would be really nice to have personally, as Process Hacker is one of the biggest tools I use that doesn't currently support this well.

### Proposed implementation details (optional)

_No response_

### PR DESCRIPTION
**Pull request by henrypp**
This is fix #1239 issue.

DONE:
- layout manager was correctly displayed when dpi changed!
- toolbar was correctly displayed when dpi changed!
- all system information graphs was correctly displayed when dpi changed!
 
TODO: 
- System Information window fonts are look like s**t when dpi changed, there is need a workaround, because graph nodes was located in different project files location (plugins, sysinformer repo) and is a little bit hard to made it right.

### CODE REVIEW

**Comment by henrypp** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by henrypp** [comment]
recheck

File: SystemInformer/hndlstat.c
Comment (jxy-s) [inline SystemInformer/hndlstat.c:158]:
```suggestion
            dpiValue = PhGetWindowDpi(hwndDlg);
```

File: SystemInformer/include/phapp.h
Comment (jxy-s) [inline SystemInformer/include/phapp.h:76]:
```suggestion
#define PH_SCALE_DPI_WINDOW(Value, Hwnd) PhMultiplyDivide(Value, PhGetWindowDpi(Hwnd), 96) // phapppub
```

File: SystemInformer/memedit.c
Comment (jxy-s) [inline SystemInformer/memedit.c:266]:
```suggestion
                dpiValue = PhGetWindowDpi(hwndDlg);
```

File: SystemInformer/prpgperf.c
Comment (jxy-s) [inline SystemInformer/prpgperf.c:156]:
```suggestion
                    dpiValue = PhGetWindowDpi(hwndDlg);
```

File: SystemInformer/memrslt.c
Comment (jxy-s) [inline SystemInformer/memrslt.c:318]: Convention elsewhere in the project avoids `SetRect`. I suggest reverting this change. And following convention elsewhere (don't use `SetRect`).

File: SystemInformer/prpgperf.c
Comment (jxy-s) [inline SystemInformer/prpgperf.c:507]: Formatting here is way off (and below).

File: SystemInformer/sysinfo.c
Comment (jxy-s) [inline SystemInformer/sysinfo.c:1670]:
```suggestion
            );
```

Comment (jxy-s) [reply SystemInformer/sysinfo.c:1670]: and throughout

File: phlib/guisup.c
Comment (jxy-s) [inline phlib/guisup.c:1227]: 🤔.. please revert this change.

File: phlib/guisup.c
Comment (jxy-s) [inline phlib/guisup.c:1238]:
```suggestion
    PhGetSizeDpiValue(&item->Rect, Manager->dpiValue, FALSE);
```

File: phlib/guisup.c
Comment (jxy-s) [inline phlib/guisup.c:1242]:
```suggestion
    PhGetSizeDpiValue(&item->Margin, Manager->dpiValue, FALSE);
```

File: phlib/include/phutil.h
Comment (jxy-s) [inline phlib/include/phutil.h:175]:
```suggestion
LONG PhGetTaskbarDpi(
    VOID
    );
```

File: phlib/include/phutil.h
Comment (jxy-s) [inline phlib/include/phutil.h:169]:
```suggestion
    );
```

Comment (jxy-s) [reply phlib/include/phutil.h:169]: and throughout

File: phlib/util.c
Comment (jxy-s) [inline phlib/util.c:167]:
```suggestion
    )
```

Comment (jxy-s) [reply phlib/util.c:167]: here and throughout

File: phlib/util.c
Comment (jxy-s) [inline phlib/util.c:166]:
```suggestion
    _In_ LONG Number,
    _In_ LONG DpiValue
```

File: phlib/util.c
Comment (jxy-s) [inline phlib/util.c:179]:
```suggestion
LONG PhGetTaskbarDpi(
    VOID
    )
```

File: phlib/util.c
Comment (jxy-s) [reply phlib/util.c:166]: please use pascal case, here and throughout

**Review by jxy-s — DISMISSED** [review]
There are a lot of unnecessary formatting and white space changes that make it really hard to review the changes. As a reviewer I can not be effective at parsing what you've actually changed vs what is just irrelevant. There are also a number of general style/formatting issues with the changes.

I'm fine with white space cleanup. I'm less okay with formatting changes. Regardless, please organize your commits and present them to reviewers in a way where they can view meaningful changes and ignore meaningless changes.

> TODO:
>
>    system information graphs was incorectly sized across displays
>    main listview icons was incorectly sized across displays

This makes me believe the changes are incomplete and can't be accepted. Although I haven't pulled your branch yet to test it yet. So it could be that it isn't problematic.

Please rebase and squash your commits. This is also a great opportunity to organize your commits in a way where reviewers have an easier time reviewing them. Also take this time to break out the white space and formatting changes into their own separate commit.

File: phlib/guisup.c
Comment (henrypp) [reply phlib/guisup.c:1227]: whats wrong? this is for correct layout manager work for for different dpi.

File: phlib/util.c
Comment (henrypp) [reply phlib/util.c:166]: I try;)

**Comment by henrypp** [comment]
> There are a lot of unnecessary formatting and white space changes that make it really hard to review the changes. As a reviewer I can not be effective at parsing what you've actually changed vs what is just irrelevant. There are also a number of general style/formatting issues with the changes.

Fixed.

> This makes me believe the changes are incomplete and can't be accepted. Although I haven't pulled your branch yet to test it yet. So it could be that it isn't problematic.

Sysinfo cpu and memory graphs was fixed, other sysinfo graphs left.

**Comment by henrypp** [comment]
Thats not me. This is VS Code doing.

sshot-001

**Comment by henrypp** [comment]
It is almost done. Now i need to know how to change sysinfo window fonts when dpi changed. Where it is?

File: SystemInformer/procprv.c
Comment (dmex) [inline SystemInformer/procprv.c:234]: Passing a window handle to PhProcessProviderInitialization breaks the abstraction between the UI layer and background provider and adds lots of unnecessary checks. 

You should move PhProcessImageListInitialization out of PhProcessImageListInitialization, calling it right before PhProcessProviderInitialization and then remove the window handle from PhProcessProviderInitialization and only pass the handle to PhProcessImageListInitialization.

File: phlib/util.c
Comment (dmex) [inline phlib/util.c:233]: Any calls to `GetProcAddress` have to be changed to `PhGetDllBaseProcedureAddress` or `PhGetProcedureAddress` because third party software hooks GetProcAddress and causes crashes.

File: phlib/util.c
Comment (dmex) [inline phlib/util.c:211]: These three GDFM, GDFW and GDFS typedefs have to be declared inline. Checkout `PhGetProcessDpiAwareness` `PhIsImmersiveProcess`, `PhGetPhysicallyInstalledSystemMemory` and `PhGetInternalWindowIcon` for some examples in `\phlib\guisup.c`

File: phlib/util.c
Comment (dmex) [inline phlib/util.c:310]: GSMFD needs to be inline the same as PhGetProcessDpiAwareness

**Comment by dmex** [comment]
> Thats not me. This is VS Code doing, but i think it is normal ;)

We can't merge this PR with hundreds of unrelated changes. Some of your commits overwrite previous commits by others with the same changes, some others overwrite comments and also update thirdparty.lib but there's no changes to the thirdparty sources?. 

You can right-click a function/source/file/code editor in visual studio and select Git > Blame Annotate to review changes and find the author/committer and those unrelated changes will overwrite the blame/review history. For example the inline blame annotation included with VS:

image

You'll probably need to copy all the files in your branch into a new directory, reset the branch to master, then copy/replace your changes and commit only the lines/files that need changing. If you have github desktop (https://desktop.github.com/) then you can select individual lines by selecting the column and commit just those lines without having to commit the entire file. 

For example the blue lines will be committed while others are ignored:
image

Other than that the PR doesn't have any other issues, good work 👍

File: phlib/util.c
Comment (dmex) [inline phlib/util.c:217]: HINSTANCE type is deprecated and should be PVOID

**Comment by dmex** [comment]
@henrypp do you know if we're also supposed to be enabling DPI scaling for built-in controls using CCM_DPISCALE? 

For example:
`SendMessage(WindowHandle, CCM_DPISCALE, TRUE, 0);`

File: SystemInformer/procprv.c
Comment (henrypp) [reply SystemInformer/procprv.c:234]: Here i did not understand where is problem

**Comment by henrypp** [comment]
> do you know if we're also supposed to be enabling DPI scaling for built-in controls using CCM_DPISCALE?

PerMonitor v2 automaticaly scale UI controls.

**Comment by henrypp** [comment]
Well done, i revert some confict changed and update whole project.

File: phlib/util.c
Comment (henrypp) [reply phlib/util.c]: did not see this in HACKING.md

**Comment by henrypp** [comment]
Well, done!

File: SystemInformer/runas.c
Comment (dmex) [inline SystemInformer/runas.c:621]: Don't change these lines

File: SystemInformer/runas.c
Comment (dmex) [inline SystemInformer/runas.c:897]: Don't change these lines

File: SystemInformer/runas.c
Comment (dmex) [inline SystemInformer/runas.c:1059]: Don't change these lines

File: SystemInformer/runas.c
Comment (dmex) [inline SystemInformer/runas.c:1395]: Don't change these lines

File: SystemInformer/runas.c
Comment (dmex) [inline SystemInformer/runas.c:1936]: Don't change these lines

File: SystemInformer/runas.c
Comment (dmex) [inline SystemInformer/runas.c:2273]: Don't change these lines

File: SystemInformer/runas.c
Comment (dmex) [inline SystemInformer/runas.c:2404]: Don't change these lines

**Review by dmex — CHANGES_REQUESTED** [review]

**Comment by dmex** [comment]
LGTM 

There's only those few issues changing comments unrelated to the PR but everything else is good 👍

File: SystemInformer/runas.c
Comment (henrypp) [reply SystemInformer/runas.c:2273]: But this is whitespaces!

File: SystemInformer/runas.c
Comment (henrypp) [reply SystemInformer/runas.c:1936]: But this is whitespacess!

File: SystemInformer/runas.c
Comment (henrypp) [reply SystemInformer/runas.c:1395]: But this is whitespacess!

File: SystemInformer/runas.c
Comment (henrypp) [reply SystemInformer/runas.c:897]: But this is whitespacess!

File: SystemInformer/runas.c
Comment (henrypp) [reply SystemInformer/runas.c:2404]: But this is whitespacess!

Comment (dmex) [reply SystemInformer/runas.c:2404]: You'll be getting blamed for these comments/lines by changing them. For example:

https://github.com/winsiderss/systeminformer/blame/44d9c6b138236b9af9812c59ccf309189fb2ad81/SystemInformer/runas.c#L2402

https://github.com/winsiderss/systeminformer/blame/44d9c6b138236b9af9812c59ccf309189fb2ad81/SystemInformer/runas.c#L1936

image

All these lines (and the others above) are somewhat security related. You don't want to get blamed for these in 5-10 years down the road if they're incorrect 😉

**Review by dmex — APPROVED** [review]

**Comment by dmex** [comment]
@henrypp 

Can I merge this PR into master or are you still making changes?

**Comment by henrypp** [comment]
@dmex

You can merge.

**Comment by dmex** [comment]
@henrypp 

This is really well done. Nice work 👍

**Pull request merged**
